vizro-core/src/vizro/models/_page.py

```diff
@@ -189,28 +189,30 @@ def _create_component_container(self, components_content):
         return component_container
 
     @staticmethod
-    def _arrange_containers(page_title, theme_switch, nav_panel, control_panel, component_container):
+    def _arrange_containers(page_title, theme_switch, navigation, control_panel, component_container):
         """Defines div container arrangement on page.
 
         To change arrangement, one has to change the order in the header, left_side and/or right_side_elements.
         """
+        nav_bar, nav_panel = navigation["nav_bar_outer"], navigation["nav_panel_outer"]
+
         _, dashboard = next(model_manager._items_with_type(Dashboard))
         dashboard_title = (
             html.Div(
                 children=[html.H2(dashboard.title), html.Hr()], className="dashboard_title", id="dashboard_title_outer"
             )
             if dashboard.title
-            else None
+            else html.Div(className="hidden")
         )
 
         header_elements = [page_title, theme_switch]
-        left_side_elements = [dashboard_title, nav_panel, control_panel]
+        nav_control_elements = [dashboard_title, nav_panel, control_panel]
+
+        nav_control_panel = html.Div(nav_control_elements, className="nav_control_panel")
```

Comment:
I think you still need to add the logic of hiding the nav_panel if there are no elements inside?

```suggestion
        nav_control_panel = html.Div(nav_control_elements, className="nav_control_panel") if any(nav_control_elements)
            else html.Div(className="hidden")
```

Reply:
Yes, I've implemented this after I pushed the changes.